app/controllers/lti_deployment_controller.rb

```diff
@@ -123,6 +123,11 @@ def redirect_login
     lti_deployment = LtiDeployment.find_or_create_by(lti_client: lti_client, external_deployment_id: deployment_id)
     session[:lti_client_id] = lti_client.id
     session[:lti_deployment_id] = lti_deployment.id
+    if decoded_token[0].key?('https://purl.imsglobal.org/spec/lti-nrps/claim/namesroleservice')
+      name_and_roles_endpoint = decoded_token[0]['https://purl.imsglobal.org/spec/lti-nrps/claim/namesroleservice']['context_memberships_url']
+      names_service = LtiService.find_or_create_by(lti_deployment: lti_deployment, service_type: 'namesroles')
```

Comment:
Can you do a `find_or_initialize_by` instead, to make `url` a required field?

Reply:
Oh and are you sure the service is `namesroles` and not `namesrole`? (Is there an official list somewhere?) Because the claim endpoint is `namesroleservice`, not `namesrolesservice`

Reply:
I think you're right, it should be `namesrole` in line with the claim, as the service is the [Names and Role Provisioning service](https://www.imsglobal.org/spec/lti-nrps/v2p0). There isn't really a master list, but I've been working from the claims/scopes canvas provides on their [LTI config page](https://canvas.instructure.com/doc/api/file.lti_dev_key_config.html#anatomy-of-a-json-configuration)